invoices/invoiceregistry.go

```diff
@@ -50,6 +50,17 @@ type HodlEvent struct {
 	AcceptHeight int32
 }
 
+// NewHodlEvent returns a new hodl event with the values specified.
+func NewHodlEvent(preimage *lntypes.Preimage, key channeldb.CircuitKey,
+	acceptHeight int32) *HodlEvent {
```

Comment:
Just syntactic sugar, but a separate constructor for settles would be nice. It makes it more explicit what the resolution is and also allows us to set a fixed ResolutionResult Settled.

Reply:
The settles constructor isn't quite as sugary because it still has to take a resolution result (since we have different settle resolution types for replays etc, and imo it's nice to have them for tests) but I think it's still nice enough to keep 🍭